### PR TITLE
Several developments for the AMOC analysis

### PR DESCRIPTION
**Pull request by ledm**
This includes several developments for the AMOC analysis:

- New regions
- New datasets
- New tool to make a generic map legend
- Revisions to the remove mask from shelve tool
- New self destruct code in mass
- New circulation code
- Ice area superceded ice extent
- yml's now have yml anchors instead of repeating the same suites list.

### CODE REVIEW

**Comment by ledm** [comment]
The test is failing because I added the `500m` layer to the `temperature.yml file`. Apparently, this fails the test?

File: bgcval2/bgcvaltools/makeMask.py [hunk 1]
```diff
@@ -216,20 +216,20 @@ def makeMask(name, newSlice, xt, xz, xy, xx, xd, debug=False):
     if newSlice == 'ignoreExtraArtics':
         return np.ma.masked_outside(xy, -50., 50.).mask
     if newSlice == 'NorthAtlanticOcean':
-        return np.ma.masked_outside(bvt.makeLonSafeArr(xx), -80.,
+        print('NorthAtlanticOcean - pre')
+        #xx = bvt.makeLonSafeArr(xx)
+        print('NorthAtlanticOcean', len(xx))
```
Comment (ledm) [inline bgcval2/bgcvaltools/makeMask.py:221]:
```suggestion
```

File: bgcval2/bgcvaltools/makeMask.py [hunk 2]
```diff
@@ -246,11 +246,36 @@ def makeMask(name, newSlice, xt, xz, xy, xx, xd, debug=False):
             xy, 60., 80.).mask
         return mx
 
+    if newSlice in ['SubtropicNorthAtlantic', 'STNA']:
+        mx = np.ma.masked_outside(xx, -80., -10.).mask + np.ma.masked_outside(
+            xy, 10., 40.).mask
+#       mx *= np.ma.masked_outside(xx, -45., 15.).mask + np.ma.masked_outside(
+#           xy, 60., 80.).mask
```
Comment (ledm) [inline bgcval2/bgcvaltools/makeMask.py:253]:
```suggestion
```

File: bgcval2/bgcvaltools/makeMask.py [hunk 3]
```diff
@@ -246,11 +246,36 @@ def makeMask(name, newSlice, xt, xz, xy, xx, xd, debug=False):
...
+        return mx
+
+    if newSlice in ['SubtropicSouthAtlantic', 'STSA']:
+        mx = np.ma.masked_outside(xx, -80., -10.).mask + np.ma.masked_outside(
+            xy, 10., 40.).mask
+#       mx *= np.ma.masked_outside(xx, -45., 15.).mask + np.ma.masked_outside(
+#           xy, 60., 80.).mask
```
Comment (ledm) [inline bgcval2/bgcvaltools/makeMask.py:260]:
```suggestion
```

File: bgcval2/bgcvaltools/makeMask.py [hunk 4]
```diff
@@ -246,11 +246,36 @@ def makeMask(name, newSlice, xt, xz, xy, xx, xd, debug=False):
...
+
     if newSlice in ['SubpolarNorthAtlantic', 'SPNA',]:
         #  Based on  SPNA region here: https://www.nature.com/articles/s43247-021-00120-y#citeas
         mx = np.ma.masked_outside(xx, -35., -10.).mask + np.ma.masked_outside(
             xy, 40., 65.).mask
         return mx
+
+#    if newSlice in ['WesternSubpolarNorthAtlantic', 'WSPNA',]:
+        
+
```
Comment (ledm) [inline bgcval2/bgcvaltools/makeMask.py:272]:
```suggestion
```

File: bgcval2/bgcvaltools/makeMask.py (hunk 1 above)
Comment (valeriupredoi) [reply bgcval2/bgcvaltools/makeMask.py:221]: yo can you maybe not do development in review mode, pls bud? I get an email everytime you "suggest" a change :rofl:

Comment (ledm) [reply bgcval2/bgcvaltools/makeMask.py:221]: Github is annoying like that. I was told there's a setting somewhere that reduces the number of emails. I don't believe it personally.

**Comment by valeriupredoi** [comment]
yep, just add the "500m" member in the list here `tests/integration/test_analysis_timeseries.py::test_load_key_file` pls bud :beer:

**Comment by valeriupredoi** [comment]
@ledm you ready for this to be merged? All fine by me :+1:

**Review by valeriupredoi — APPROVED** [review]

**Pull request merged**